app/assets/javascripts/deploys.js

```diff
@@ -34,26 +34,44 @@ $(function() {
     });
 });
 
-function update_progress_bar(progress_data) {
+function update_progress_bar(progress_data, error) {
+  const bar_id = "#progress-bar";
+  const progress_bar_id = "div" + bar_id;
+
+  $(progress_bar_id).css("width", progress_data.progress + "%");
```

Comment:
Old-skool jQuery top tip: instead of putting a selector in a string var and then invoking the DOM traversal, you can store the DOM element itself, _after_ having traversed the DOM in search of it, e.g.:
```javascript
const $progress_bar = $("div#progress-bar");
$progress_bar.css( /*etc, etc...*/

```
This way you can avoid repeating the DOM traversal every time.

Note that it is conventional to name such variables, referencing jQuery DOM elements, with a `$` at the start (to let readers know that variable exposes the jQuery API).

Reply:
Thanks! I will try to make these improvements

Reply:
@stefanotorresi Improvements done in https://github.com/SUSE/blue-horizon/pull/17/commits/b506cb410b45b2932edee1f31643f4e7fa1bc02f
The code looks much neater. Thanks for the suggestions

Reply:
excellent!! 👍🏼